From qmail-return-90430-nelson-l-6ca615d95fcb76f5bb3afdc1a6ef9814=crynwr.com@list.cr.yp.to Mon Aug 05 12:02:02 2002
X-VM-v5-Data: ([nil nil nil nil nil nil nil nil nil]
	["2277" "Mon" "5" "August" "2002" "19:51:50" "+0800" "Adrian Ho" "aho-djb-djbdns@03s.net" nil "58" "Re: Odd registrar behaviour" "^From:" nil nil "8" nil nil nil nil nil]
	nil)
Return-Path: <qmail-return-90430-nelson-l-6ca615d95fcb76f5bb3afdc1a6ef9814=crynwr.com@list.cr.yp.to>
Delivered-To: nelson@desk.crynwr.com
Received: (qmail 16480 invoked by uid 501); 5 Aug 2002 12:02:02 -0000
Delivered-To: nelson-l-6ca615d95fcb76f5bb3afdc1a6ef9814@desk.crynwr.com
Received: (qmail 16477 invoked from network); 5 Aug 2002 12:02:02 -0000
Received: from ns1.crynwr.com (HELO ns.crynwr.com) (192.203.178.14)
  by desk.crynwr.com with SMTP; 5 Aug 2002 12:02:02 -0000
Received: (qmail 20090 invoked by uid 500); 5 Aug 2002 12:00:34 -0000
Delivered-To: nelson-l-6ca615d95fcb76f5bb3afdc1a6ef9814@crynwr.com
Received: (qmail 20086 invoked from network); 5 Aug 2002 12:00:33 -0000
Received: from muncher.math.uic.edu (131.193.178.181)
  by pdam.crynwr.com with SMTP; 5 Aug 2002 12:00:32 -0000
Received: (qmail 30998 invoked by uid 1002); 5 Aug 2002 11:58:09 -0000
Mailing-List: contact qmail-help@list.cr.yp.to; run by ezmlm
Precedence: bulk
Delivered-To: mailing list qmail@list.cr.yp.to
Received: (qmail 11404 invoked from network); 5 Aug 2002 11:51:26 -0000
Received: from cm22.omega47.scvmaxonline.com.sg (HELO svr1.03s.net) (218.186.47.22)
  by muncher.math.uic.edu with SMTP; 5 Aug 2002 11:51:26 -0000
Received: (qmail 5331 invoked by uid 501); 5 Aug 2002 11:51:52 -0000
X-pymsgauth-token: b3430d9322a7425fb5b15ecf7fc8b0e994152e1e
Message-ID: <20020805195150.A21092@svr1.03s.net>
Mail-Followup-To: qmail@list.cr.yp.to
References: <20020802144312.17c5cf4b.lars@unet.net.ph> <20020802081526.GG384@straylight.oblivion.bg> <05e601c23a74$c722d000$cd4751d1@primedata.org> <m3vg6sn9p9.fsf@multivac.cwru.edu> <009401c23b2d$e46f6d30$cd4751d1@primedata.org> <m3r8hexme0.fsf@multivac.cwru.edu> <20020805091156.00936977.lars@unet.net.ph> <20020805162406.53638cee.lars@unet.net.ph> <20020805175801.A30645@svr1.03s.net> <20020805182737.53267b58.lars@unet.net.ph>
Mime-Version: 1.0
Content-Type: multipart/mixed; boundary="5vNYLRcllDrimb99"
Content-Disposition: inline
User-Agent: Mutt/1.3.15i
In-Reply-To: <20020805182737.53267b58.lars@unet.net.ph>; from lars@unet.net.ph on Mon, Aug 05, 2002 at 06:27:37PM +0800
From: Adrian Ho <aho-djb-djbdns@03s.net>
To: dns@list.cr.yp.to, qmail@list.cr.yp.to
Subject: Re: Odd registrar behaviour
Date: Mon, 5 Aug 2002 19:51:50 +0800


--5vNYLRcllDrimb99
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline

[Followups to qmail@list.cr.yp.to.  For those just tuning in, the topic
under discussion is a catch-all SMTP server for unassigned .ph domains.
Specifically, this SMTP server returns a 574 banner and drops the
connection immediately; the latter is prohibited under RFC 2821.]

On Mon, Aug 05, 2002 at 06:27:37PM +0800, Lars Hansson wrote:
> On Mon, 5 Aug 2002 17:58:01 +0800 Adrian Ho <aho-djb-djbdns@03s.net> wrote:
> > A "friendlier" behavior for x.ph would be to respond with a 220 banner,
> > then 503 every subsequent command till a QUIT.  Sorta like rblsmtpd with
> > RBLSMTPD="-Relay not authorized".
> 
> Actually, that doesnt seem to work either.

Well, it doesn't work for qmail-remote, yes.

> Sending a 220 banner and replying 250 to all subsequent commands except
> for "rcpt to" will work if "rcpt to" returns 503.

Actually, a 503 to MAIL FROM would be sufficient for qmail-remote.

That said, AFAICT qmail-remote's handling of this scenario deviates from
2821 (to be fair, the latter was of course published after qmail-1.03).
I've attached a trivial patch to qmail-remote.c that fixes this.

-- 
Adrian Ho    Tinker, Drifter, Fixer, Bum   aho-djb-djbdns@03s.net
ListArchive: <http://marc.theaimsgroup.com/?l=djbdns>
Useful URLs: <http://cr.yp.to/djbdns.html> <http://tinydns.org/>
             <http://www.lifewithdjbdns.org/> <http://djbdns.faqts.com/>

From: Adrian Ho <aho-djb-qmail@03s.net>
To: qmail@list.cr.yp.to
Subject: Re: Odd registrar behaviour
Date: Fri, 16 Aug 2002 22:56:38 +0800

On Fri, Aug 16, 2002 at 04:03:30PM +0800, Lars Hansson wrote:
> > I'll post the final corrected patch /after/ I've tested it myself.

OK, the final corrected version is appended.  Apologies for all
inconvenience caused.

> On a related note, anyone knows how to make it work with the TLS patches?
> (Vermeulen's patches)

Frederik Vermeulen certainly would.

--- qmail-1.03/qmail-remote.c.orig	Mon Jun 15 18:53:16 1998
+++ qmail-1.03/qmail-remote.c	Tue Aug  6 22:43:12 2002
@@ -222,13 +222,17 @@
   int flagbother;
   int i;
  
-  if (smtpcode() != 220) quit("ZConnected to "," but greeting failed");
+  code = smtpcode();
+  if (code >= 500) quit("DConnected to "," but greeting failed");
+  if (code != 220) quit("ZConnected to "," but greeting failed");
  
   substdio_puts(&smtpto,"HELO ");
   substdio_put(&smtpto,helohost.s,helohost.len);
   substdio_puts(&smtpto,"\r\n");
   substdio_flush(&smtpto);
-  if (smtpcode() != 250) quit("ZConnected to "," but my name was rejected");
+  code = smtpcode();
+  if (code >= 500) quit("DConnected to "," but my name was rejected");
+  if (code != 250) quit("ZConnected to "," but my name was rejected");
  
   substdio_puts(&smtpto,"MAIL FROM:<");
   substdio_put(&smtpto,sender.s,sender.len);

-- 
Adrian Ho    Tinker, Drifter, Fixer, Bum   aho-djb-qmail@03s.net
ListArchive: <http://marc.theaimsgroup.com/?l=qmail>
Useful URLs: <http://cr.yp.to/qmail.html> <http://www.qmail.org>
             <http://www.lifewithqmail.org/> <http://qmail.faqts.com/>
